src/ucp/tag/rndv.c

```diff
@@ -82,8 +134,12 @@ static size_t ucp_tag_rndv_rts_pack(void *dest, void *arg)
     rndv_rts_hdr->size             = sreq->send.length;
     if (UCP_DT_IS_CONTIG(sreq->send.datatype)) {
         rndv_rts_hdr->address = (uintptr_t) sreq->send.buffer;
-        if (ucp_ep_is_rndv_lane_present(ep)) {
-            packed_len += ucp_tag_rndv_pack_rkey(sreq, ucp_ep_get_rndv_get_lane(ep),
+        if (ucp_ep_is_rndv_mrail_present(ep)) {
+            packed_len += ucp_tag_rndv_pack_mrail_rkeys(sreq,
```

Comment:
right, and moreover - refactor all rndv processing, because it is spaghetti logic for rndv/tag offload/mrail. I tried to minimize impact to original code to avoid regressions

Reply:
need to fix this